File: Data Structures & Algorithms/reverse-bits/submission-0.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {

         int i = 0;
         int ans = 0;
         while(i < 32){
            int bit = (n & 1) == 1 ? 1 : 0;
            //cout<<bit<<" at "<<i<<endl;
            if(bit == 1){
                ans = (ans << 1) + 1;
            }else{
                ans = ans << 1;
            }

            n = n >> 1;

            i++;
         }

         return ans;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        nums.sort()

        count = 1
        pq = []

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                count += 1
            else:
                if len(pq) < k:
                    heapq.heappush(pq, (count, nums[i - 1]))
                else:
                    if pq[0][0] < count:
                        heapq.heappop(pq)
                        heapq.heappush(pq, (count, nums[i - 1]))

                count = 1

        if len(pq) < k:
            heapq.heappush(pq, (count, nums[-1]))
        elif pq[0][0] < count:
            heapq.heappop(pq)
            heapq.heappush(pq, (count, nums[-1]))

        ans = []

        while pq:
            count, num = heapq.heappop(pq)
            ans.append(num)

        return ans

